### DISCUSSION TITLE
Does publication throughput from Statistics Module collect data also from builtin DataWriters?

### DISCUSSION THREAD

**Discussion by save9402** [eProsima/Fast-DDS#4039] [Q&A] [answered]
Hi everyone,
I'm writing a simple application using FastDDS, and i am using FastDDS Statistics Backend library to collect statistics. 
I have a publisher that send 34 byte each 5 milliseconds, so that means 6800 B/s, but the "publication throughput" statistics measure on average around 8650 B/s. The documentation (https://fast-dds.docs.eprosima.com/en/latest/fastdds/statistics/dds_layer/topic_names.html#publication-throughput-topic) says the statistics "collects the amount of data (expressed in B/s) that is being sent by each DataWriter", but does NOT specify "This topic does not apply to builtin (related to Discovery) and statistics DataWriters" as for other statistic topics like "heartbeat count" and others. Does that means that the "publication throughput" statistics collects also  Discovery and Statistics-related data from DataWriters?

**Comment by EduPonz** [comment, marked as answer]
Hi @save9402,

You are correct, that statistic is reported for all DataWriters (including the discovery related ones). In any case, on the backend, you can query based on the specific writer should you want to do so. Also, it is not possible in CDR serialization to have a payload of 34 bytes, as it will always get aligned to the next multiple of 4, in this case 36, which would result in 7200 B/s

**Reply by save9402** [reply]
Hi, thank you very much for your fast reply :)
